pkg/datasources/database_role.go

```diff
@@ -0,0 +1,71 @@
+package datasources
+
+import (
+	"context"
+	"log"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/internal/provider"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var databaseRoleSchema = map[string]*schema.Schema{
```

Comment:
docs have to be generated for this one, so just run `make pre-push` and the documentation should be generated. If any error pops up, refer to the [contribution guidelines](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/CONTRIBUTING.md#making-a-contribution).